### PR TITLE
[data] Add tip for validating GCS permissions

### PR DESCRIPTION
**Pull request by matthewdeng**
## Why are these changes needed?

When reading from (Google) Cloud Storage, users may run into errors that are raised by the filesystem, e.g. if the user does not have permissions. While it is preferable to raise a descriptive error message here, it may be difficult to capture all the possible error messages raised by the user provided filesystem, which may also change across different versions. This tip is meant to be a catch all to provide users with a quick debugging tip.

image

Happy to add similar notes for the other providers (S3, HDFS, Azure) if this looks good. (**Note:** These will have different tips depending if they follow the `arrow` or `fsspec` interface.)

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/data/doc_code/creating_datasets.py [hunk 1]
```diff
@@ -568,13 +568,20 @@
...
+path = "gs://path/to/file.parquet"
+filesystem = gcsfs.GCSFileSystem(project="my-google-project")
+ds = ray.data.read_parquet(path, filesystem=filesystem)
 # __read_parquet_gcs_end__
 # fmt: on
 
+
+# fmt: off
+# __validate_parquet_gcs_begin__
+print(filesystem.open(path))
```
Comment (c21) [inline doc/source/data/doc_code/creating_datasets.py:580]: shall we also add a link to gcsfs examples - https://gcsfs.readthedocs.io/en/latest/#examples ? So users can try out more APIs to unconver issues with gcsfs? Something like

```
See documentation of GCSFS for more examples - https://gcsfs.readthedocs.io/en/latest/#examples
```

File: doc/source/data/doc_code/creating_datasets.py [hunk 2]
```diff
@@ -568,13 +568,20 @@
...
+# <File-like object GCSFileSystem, path/to/file.parquet>
```
Comment (c21) [inline doc/source/data/doc_code/creating_datasets.py:581]: Another common permission issue is to list directory or file. Shall we also add a print for 

```
print(filesystem.ls(path))
```

**Review by c21 — APPROVED** [review]
Thanks @matthewdeng!

File: doc/source/data/doc_code/creating_datasets.py (hunk 1 above)
Comment (matthewdeng) [reply doc/source/data/doc_code/creating_datasets.py:580]: Oh yes this is a great point, will add.

File: doc/source/data/creating-datasets.rst [hunk 3]
```diff
@@ -293,6 +293,17 @@ are supported for each of these storage systems.
     :start-after: __read_parquet_gcs_begin__
     :end-before: __read_parquet_gcs_end__
 
+  .. tip::
+    To verify that your GCP project and credentials are set up, you can validate
```
Comment (angelinalg) [inline doc/source/data/creating-datasets.rst:297]:
```suggestion
    To verify that your GCP project and credentials are set up, validate
```

File: doc/source/data/creating-datasets.rst [hunk 4]
```diff
@@ -293,6 +293,17 @@ are supported for each of these storage systems.
...
+    that your GCS `filesystem` has the right permissions to read the input `path`.
```
Comment (angelinalg) [inline doc/source/data/creating-datasets.rst:298]:
```suggestion
    that the GCS `filesystem` has the right permissions to read the input `path`.
```

File: doc/source/data/doc_code/creating_datasets.py [hunk 5]
```diff
@@ -568,13 +568,22 @@
 # GCSFileSystem.
 # NOTE: This example is not runnable as-is; you'll need to point it at your GCS bucket
```
Comment (angelinalg) [inline doc/source/data/doc_code/creating_datasets.py:569]:
```suggestion
# NOTE: This example is not runnable as-is; you need to point it at your GCS bucket
```

File: doc/source/data/creating-datasets.rst [hunk 6]
```diff
@@ -293,6 +293,17 @@ are supported for each of these storage systems.
...
+
+    .. literalinclude:: ./doc_code/creating_datasets.py
+      :language: python
+      :start-after: __validate_parquet_gcs_begin__
+      :end-before: __validate_parquet_gcs_end__
+
+    For more examples, see `GCSFS Documentation <https://gcsfs.readthedocs.io/en/latest/#examples>`__.
```
Comment (angelinalg) [inline doc/source/data/creating-datasets.rst:305]:
```suggestion
    For more examples, see the `GCSFS Documentation <https://gcsfs.readthedocs.io/en/latest/#examples>`__.
```

**Review by angelinalg — CHANGES_REQUESTED** [review]
LGTM

**Review by angelinalg — APPROVED** [review]
LGTM

File: doc/source/data/doc_code/creating_datasets.py [hunk 7]
```diff
@@ -566,15 +566,24 @@
 
 # Create a tabular Dataset by reading a Parquet file from GCS, passing the configured
 # GCSFileSystem.
-# NOTE: This example is not runnable as-is; you'll need to point it at your GCS bucket
+# NOTE: This example is not runnable as-is; you need to point it at your GCS bucket
```
Comment (angelinalg) [inline doc/source/data/doc_code/creating_datasets.py:569]:
```suggestion
# NOTE: This example is not runnable as-is; you need to point it to your GCS bucket
```

**Review by clarkzinzow — APPROVED** [review]
Nice!

**Pull request merged**